packages/models-library/src/models_library/projects.py

```diff
@@ -0,0 +1,193 @@
+import sys
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+from pydantic import UUID4, BaseModel, EmailStr, Extra, Field, HttpUrl, constr
+
+from .services import KEY_RE, PROPERTY_KEY_RE, VERSION_RE
+
+current_file = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve()
+
+DATE_RE = r"\d{4}-(12|11|10|0?[1-9])-(31|30|[0-2]?\d)T(2[0-3]|1\d|0?[0-9])(:(\d|[0-5]\d)){2}(\.\d{3})?Z"
+
+
+class Connection(BaseModel):
+    nodeUuid: Optional[str]
+    output: Optional[str]

```

Comment:
**Note:** this comment is valid for all the below defined classes in this module.

Could you please add a few words explaining more exactly where these definitions were first intended to be used?
I am not sure if the module osparc service name and maybe API or database collections might do the trick.
As a developer I would like to have an idea where these were defined and for which purpose were intended.

Reply:
right. so I was not the original writer of this. but I see your point and will fix it to make it more understandable. but I don't like comments if not necessary. let me try to improve the naming.

Reply:
ok please check again and let me know if this is still unclear... actually added the missing parts